source/libraries/CompileSolidity.ts

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env node
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as solc from 'solc';
+
+class SolidityContractCompiler {
+    private compiler: any = solc;
+    private contractInputDirectoryPath: string;
+    private contractOutputDirectoryPath: string;
+    private contractOutputFileName: string;
+
+    public constructor(contractInputDirectoryPath: string, contractOutputDirectoryPath: string, contractOutputFileName: string) {
+        this.contractInputDirectoryPath = contractInputDirectoryPath;
+        this.contractOutputDirectoryPath = contractOutputDirectoryPath;
+        this.contractOutputFileName = contractOutputFileName;
+    }
+
+    public compileContracts() {
+        // Compile all contracts in this.inputDirectoryPath
+        let inputJSON: any = this.generateCompilerInput();
+        let compilerOutput: any = this.compiler.compile(inputJSON);
+        if (compilerOutput.errors.length > 0) {
+            console.log("The compiler encountered the following warnings/errors:");
+            for (let error of compilerOutput.errors) {
+                console.log(error);
+            }
+            // Do not return automatically in case there are only warnings and no errors
+        }
+
+        // Create output directory (if it doesn't exist) and save contract bytecodes to single file
+        this.createOutputDirectory();
+        let contractOutputFilePath: string = this.contractOutputDirectoryPath + "/" + this.contractOutputFileName;
+        let wstream: any = fs.createWriteStream(contractOutputFilePath);
+        for (let contract in compilerOutput.contracts) {
+            wstream.write(compilerOutput.contracts[contract].bytecode);
+        }
+
+        console.log("Saved " + contractOutputFilePath);
+    }
+
+    private getContractFilePaths(directoryPath){
```

Comment:
Consider using an NPM module rather than rolling your own directory listing function.  I haven't used it, but https://www.npmjs.com/package/recursive-readdir looks like it does what we want.  It is also async friendly, which is nice.  However, it appears that it returns absolute file pahts, so there may be value in looking to see if there is a better package that returns relative file paths so we don't have to strip off the base.

Reply:
Ok, I wasn't sure if I should try to minimize dependencies on other modules.  I'll look into incorporating recursive-readdir.